### PR TITLE
nprofile UI support

### PR DESCRIPTION
**Pull request by shaibearary**

### CODE REVIEW

File: UI/message.ts [hunk 1]
```diff
@@ -79,6 +105,7 @@ export type ContentItem = {
     pubkey: string;
     start: number;
     end: number;
+    relays?: string[] | undefined;
```
Comment (BlowaterNostr) [inline UI/message.ts:108]: PR Blocker: `field: type | undefined` and `field?: type` are the same.

File: UI/message.ts [hunk 2]
```diff
@@ -58,6 +61,29 @@ function* match(regex: RegExp, content: string, type: ItemType): Generator<Conte
...
+            const decoded_nProfile = NostrProfile.decode(bech32);
+            if (decoded_nProfile instanceof Error) {
+                // ignore
+            } else {
+                const pubkey = PublicKey.FromBech32(decoded_nProfile.pubkey.hex);
+                if (pubkey instanceof Error) { //ignore
+                } else {
+                    yield {
+                        type: "npub",
+                        pubkey: pubkey.hex,
```
Comment (BlowaterNostr) [inline UI/message.ts:80]: PR Blocker: better to use `PublicKey` as the type.

File: UI/message.ts [hunk 3]
```diff
@@ -58,6 +61,29 @@ function* match(regex: RegExp, content: string, type: ItemType): Generator<Conte
...
+            }
+            const decoded_nProfile = NostrProfile.decode(bech32);
+            if (decoded_nProfile instanceof Error) {
+                // ignore
+            } else {
+                const pubkey = PublicKey.FromBech32(decoded_nProfile.pubkey.hex);
+                if (pubkey instanceof Error) { //ignore
+                } else {
+                    yield {
+                        type: "npub",
```
Comment (BlowaterNostr) [inline UI/message.ts:79]: Not a blocker. Why not `nprofile`?

**Review by BlowaterNostr — CHANGES_REQUESTED** [review]

File: UI/message.ts (hunk 3 above)
Comment (shaibearary) [reply UI/message.ts:79]: my thought is, npub and nprofile is same ItemType when rendering
all pubkey + relays(optional)
the difference is the decode process, but the decode result is same type

Comment (BlowaterNostr) [reply UI/message.ts:79]: Well thought

File: UI/message.ts [hunk 4]
```diff
@@ -10,6 +10,9 @@ export function* parseContent(content: string) {
     // npubs
     yield* match(/(nostr:)?npub[0-9a-z]{59}/g, content, "npub");
 
+    //nprofile
+    yield* match(/(nostr:)?nprofile[^ ]+/g, content, "nprofile");
```
Comment (BlowaterNostr) [inline UI/message.ts:14]: image
This is apparently wrong and your unit tests failed to catch it

**Review by BlowaterNostr — CHANGES_REQUESTED** [review]

File: UI/message.ts (hunk 4 above)
Comment (BlowaterNostr) [reply UI/message.ts:14]: What you need to match is characters that are part of bech32 encoding instead of "not space"

File: UI/message.ts [hunk 5]
```diff
@@ -58,6 +61,29 @@ function* match(regex: RegExp, content: string, type: ItemType): Generator<Conte
...
+            if (match[0].startsWith("nostr:")) {
+                bech32 = content.slice(urlStartPosition + 6, urlEndPosition + 1);
+            } else {
+                bech32 = content.slice(urlStartPosition, urlEndPosition + 1);
+            }
+            const decoded_nProfile = NostrProfile.decode(bech32);
+            if (decoded_nProfile instanceof Error) {
+                // ignore
+            } else {
+                const pubkey = PublicKey.FromBech32(decoded_nProfile.pubkey.hex);
```
Comment (foodstr) [inline UI/message.ts:75]: should be PublicKey.FromBech32(decoded_nProfile.pubkey.bech32());

Comment (foodstr) [reply UI/message.ts:75]: or PublicKey.FromHex(decoded_nProfile.pubkey.hex);

Comment (BlowaterNostr) [reply UI/message.ts:75]: Should be `decoded_nProfile.pubkey`

Comment (foodstr) [reply UI/message.ts:75]: ummm, lol

**Pull request merged**